src/main/java/com/saralein/server/router/ServerRouter.java

```diff
@@ -1,40 +1,49 @@
 package com.saralein.server.router;
 
+import com.saralein.server.Controller.DirectoryController;
+import com.saralein.server.Controller.FileController;
+import com.saralein.server.Controller.NotFoundController;
 import com.saralein.server.request.Request;
 import com.saralein.server.response.*;
-import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 
 public class ServerRouter implements Router {
     private final FileHelper fileHelper;
-    private File resource = null;
+    private final Routes routes;
+    private Path resource = null;
     private boolean resourceExists = false;
     private boolean resourceIsDirectory = false;
 
-    public ServerRouter(FileHelper fileHelper) {
+    public ServerRouter(Routes routes, FileHelper fileHelper) {
+        this.routes = routes;
         this.fileHelper = fileHelper;
     }
 
-    public byte[] resolveRequest(Request request) {
+    public Response resolveRequest(Request request) {
         resourceStatus(request);
         return route(request);
     }
 
     private void resourceStatus(Request request) {
         String resourceUri = fileHelper.createAbsolutePath(request.getUri());
 
-        resource = new File(resourceUri);
+        resource = Paths.get(resourceUri);
 
-        resourceExists = resource.exists();
-        resourceIsDirectory = resource.isDirectory();
+        resourceExists = Files.exists(resource);
+        resourceIsDirectory = Files.isDirectory(resource);
     }
 
-    private byte[] route(Request request) {
-        if (!resourceExists) {
-            return new NotFoundResponse().createResponse();
+    private Response route(Request request) {
+        if (routes.isRoute(request.getUri())) {
+            return routes.getController(request.getUri()).createResponse();
+        } else if (!resourceExists) {
+            return new NotFoundController().createResponse();
         } else if (resourceIsDirectory) {
-            return new DirectoryResponse(resource, fileHelper).createResponse();
+            return new DirectoryController(resource, fileHelper).createResponse();
         } else {
-            return new FileResponse(request, resource, fileHelper).createResponse();
+            return new FileController(request, resource, fileHelper).createResponse();
         }
```

Comment:
Is there any reason why these controllers can't conform to the same interface as the other controllers? Like can they just be injected into that list of routes?

Reply:
Yes, I can do that.  I might do it on the current branch I'm working on though since I'm working around the same area a lot.

Reply:
OK